[
  {
    "package": {
      "name": "urllib3",
      "version": "1.26.18",
      "ecosystem": "PyPI"
    },
    "dependency_groups": [
      "buku-requirements"
    ],
    "vulnerabilities": [
      {
        "modified": "2024-06-17T22:12:25Z",
        "published": "2024-06-17T21:37:20Z",
        "schema_version": "1.6.0",
        "id": "GHSA-34jh-p97f-mpxf",
        "aliases": [
          "CVE-2024-37891"
        ],
        "summary": "urllib3's Proxy-Authorization request header isn't stripped during cross-origin redirects ",
        "details": "When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected.\n\nHowever, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects.\n\nBecause this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident.\n\nUsers should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach.\n\n## Affected usages\n\nWe believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited:\n\n* Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support.\n* Not disabling HTTP redirects.\n* Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin.\n\n## Remediation\n\n* Using the `Proxy-Authorization` header with urllib3's `ProxyManager`.\n* Disabling HTTP redirects using `redirects=False` when sending requests.\n* Not using the `Proxy-Authorization` header.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "urllib3",
              "purl": "pkg:pypi/urllib3"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "1.26.19"
                  }
                ]
              }
            ],
            "versions": [
              "0.2",
              "0.3",
              "0.3.1",
              "0.4.0",
              "0.4.1",
              "1.0",
              "1.0.1",
              "1.0.2",
              "1.1",
              "1.10",
              "1.10.1",
              "1.10.2",
              "1.10.3",
              "1.10.4",
              "1.11",
              "1.12",
              "1.13",
              "1.13.1",
              "1.14",
              "1.15",
              "1.15.1",
              "1.16",
              "1.17",
              "1.18",
              "1.18.1",
              "1.19",
              "1.19.1",
              "1.2",
              "1.2.1",
              "1.2.2",
              "1.20",
              "1.21",
              "1.21.1",
              "1.22",
              "1.23",
              "1.24",
              "1.24.1",
              "1.24.2",
              "1.24.3",
              "1.25",
              "1.25.1",
              "1.25.10",
              "1.25.11",
              "1.25.2",
              "1.25.3",
              "1.25.4",
              "1.25.5",
              "1.25.6",
              "1.25.7",
              "1.25.8",
              "1.25.9",
              "1.26.0",
              "1.26.1",
              "1.26.10",
              "1.26.11",
              "1.26.12",
              "1.26.13",
              "1.26.14",
              "1.26.15",
              "1.26.16",
              "1.26.17",
              "1.26.18",
              "1.26.2",
              "1.26.3",
              "1.26.4",
              "1.26.5",
              "1.26.6",
              "1.26.7",
              "1.26.8",
              "1.26.9",
              "1.3",
              "1.4",
              "1.5",
              "1.6",
              "1.7",
              "1.7.1",
              "1.8",
              "1.8.2",
              "1.8.3",
              "1.9",
              "1.9.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/06/GHSA-34jh-p97f-mpxf/GHSA-34jh-p97f-mpxf.json"
            }
          },
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "urllib3",
              "purl": "pkg:pypi/urllib3"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "2.0.0"
                  },
                  {
                    "fixed": "2.2.2"
                  }
                ]
              }
            ],
            "versions": [
              "2.0.0",
              "2.0.1",
              "2.0.2",
              "2.0.3",
              "2.0.4",
              "2.0.5",
              "2.0.6",
              "2.0.7",
              "2.1.0",
              "2.2.0",
              "2.2.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/06/GHSA-34jh-p97f-mpxf/GHSA-34jh-p97f-mpxf.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N"
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf"
          },
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-37891"
          },
          {
            "type": "WEB",
            "url": "https://github.com/urllib3/urllib3/commit/40b6d1605814dd1db0a46e202d6e56f2e4c9a468"
          },
          {
            "type": "WEB",
            "url": "https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/urllib3/urllib3"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-669"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2024-06-17T21:37:20Z",
          "nvd_published_at": "2024-06-17T20:15:13Z",
          "severity": "MODERATE"
        }
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-34jh-p97f-mpxf"
        ],
        "aliases": [
          "CVE-2024-37891",
          "GHSA-34jh-p97f-mpxf"
        ],
        "max_severity": "4.4"
      }
    ]
  },
  {
    "package": {
      "name": "werkzeug",
      "version": "2.3.8",
      "ecosystem": "PyPI"
    },
    "dependency_groups": [
      "buku-requirements"
    ],
    "vulnerabilities": [
      {
        "modified": "2024-06-14T16:19:02Z",
        "published": "2024-05-06T14:21:27Z",
        "schema_version": "1.6.0",
        "id": "GHSA-2g68-c3qc-8985",
        "aliases": [
          "CVE-2024-34069"
        ],
        "summary": "Werkzeug debugger vulnerable to remote execution when interacting with attacker controlled domain",
        "details": "The debugger in affected versions of Werkzeug can allow an attacker to execute code on a developer's machine under some circumstances. This requires the attacker to get the developer to interact with a domain and subdomain they control, and enter the debugger PIN, but if they are successful it allows access to the debugger even if it is only running on localhost. This also requires the attacker to guess a URL in the developer's application that will trigger the debugger.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "werkzeug",
              "purl": "pkg:pypi/werkzeug"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "3.0.3"
                  }
                ]
              }
            ],
            "versions": [
              "0.1",
              "0.10",
              "0.10.1",
              "0.10.2",
              "0.10.3",
              "0.10.4",
              "0.11",
              "0.11.1",
              "0.11.10",
              "0.11.11",
              "0.11.12",
              "0.11.13",
              "0.11.14",
              "0.11.15",
              "0.11.2",
              "0.11.3",
              "0.11.4",
              "0.11.5",
              "0.11.6",
              "0.11.7",
              "0.11.8",
              "0.11.9",
              "0.12",
              "0.12.1",
              "0.12.2",
              "0.13",
              "0.14",
              "0.14.1",
              "0.15.0",
              "0.15.1",
              "0.15.2",
              "0.15.3",
              "0.15.4",
              "0.15.5",
              "0.15.6",
              "0.16.0",
              "0.16.1",
              "0.2",
              "0.3",
              "0.3.1",
              "0.4",
              "0.4.1",
              "0.5",
              "0.5.1",
              "0.6",
              "0.6.1",
              "0.6.2",
              "0.7",
              "0.7.1",
              "0.7.2",
              "0.8",
              "0.8.1",
              "0.8.2",
              "0.8.3",
              "0.9",
              "0.9.1",
              "0.9.2",
              "0.9.3",
              "0.9.4",
              "0.9.5",
              "0.9.6",
              "1.0.0",
              "1.0.0rc1",
              "1.0.1",
              "2.0.0",
              "2.0.0rc1",
              "2.0.0rc2",
              "2.0.0rc3",
              "2.0.0rc4",
              "2.0.0rc5",
              "2.0.1",
              "2.0.2",
              "2.0.3",
              "2.1.0",
              "2.1.1",
              "2.1.2",
              "2.2.0",
              "2.2.0a1",
              "2.2.1",
              "2.2.2",
              "2.2.3",
              "2.3.0",
              "2.3.1",
              "2.3.2",
              "2.3.3",
              "2.3.4",
              "2.3.5",
              "2.3.6",
              "2.3.7",
              "2.3.8",
              "3.0.0",
              "3.0.1",
              "3.0.2"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/05/GHSA-2g68-c3qc-8985/GHSA-2g68-c3qc-8985.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H"
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985"
          },
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34069"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pallets/werkzeug"
          },
          {
            "type": "WEB",
            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/H4SH32AM3CTPMAAEOIDAN7VU565LO4IR"
          },
          {
            "type": "WEB",
            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HFERFN7PINV4MOGMGA3DPIXJPDCYOEJZ"
          },
          {
            "type": "WEB",
            "url": "https://security.netapp.com/advisory/ntap-20240614-0004"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-352"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2024-05-06T14:21:27Z",
          "nvd_published_at": "2024-05-06T15:15:23Z",
          "severity": "HIGH"
        }
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-2g68-c3qc-8985"
        ],
        "aliases": [
          "CVE-2024-34069",
          "GHSA-2g68-c3qc-8985"
        ],
        "max_severity": "7.5"
      }
    ]
  }
]